client/express-checkout/utils/checkPaymentMethodIsAvailable.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import ReactDOM from 'react-dom';
+import { ExpressCheckoutElement, Elements } from '@stripe/react-stripe-js';
+import { memoize } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { isLinkEnabled } from 'wcpay/checkout/utils/upe';
+import request from 'wcpay/checkout/utils/request';
+import WCPayAPI from 'wcpay/checkout/api';
+import { getUPEConfig } from 'wcpay/utils/checkout';
+
+export const checkPaymentMethodIsAvailable = memoize(
+	( paymentMethod, cart, resolve ) => {
+		const root = ReactDOM.createRoot(
```

Comment:
I am getting the following warning from this line. I'm not sure why it's not getting flagged in `WooPayExpressCheckoutButtonContainer` also?

```
Warning: You are importing createRoot from "react-dom" which is not supported. You should instead import it from "react-dom/client".
    at span
    at WooPayExpressCheckoutButtonContainer (webpack-internal:///./client/checkout/woopay/express-button/woopay-express-checkout-payment-method.js:39:73)
```

Reply:
I see it is being thrown from the Container in that message I copied, when I was looking reproduce, it was coming from the `checkPaymentMethodIsAvailable` function?

Reply:
"I don't get this error, @bborman22. On which page are you getting this warning?

Reply:
It's odd we haven't noticed this before, as this is the only way we're importing ReactDOM across the entire plugin.

Reply:
Yeah it's really odd because like in my original mix up above, it seems to come intermittently. I also see it happening for WC Core blocks-checkout. I see this warning on the blocks checkout page.

![Screenshot 2024-09-06 at 9 47 03 AM](https://github.com/user-attachments/assets/d2ed70d2-dd11-41ab-a770-0a9d5a4801e5)

It doesn't seem to be causing any issues and seems to work as expected? I also noticed that this error should only be coming from `react-dom` v18+ and in WooPayments we're using 17.0.2?

Given the following I think we can ignore this for now:
- You're not seeing the same error.
- We are using createRoot this way everywhere.
- The warning shouldn't even happen in our version of react-dom and `createRoot` [isn't exported from `react-dom/client` until v18](https://github.com/facebook/react/blob/main/CHANGELOG.md#react-dom-client).

We can ignore this for now.